openstack_controller/datadog_checks/openstack_controller/api/api_rest.py

```diff
@@ -381,11 +381,29 @@ def use_legacy_nodes_resource(microversion):
         return self.make_paginated_request(url, 'nodes', 'uuid', params=params)
 
     def get_baremetal_conductors(self):
-        response = self.http.get(
-            '{}/v1/conductors'.format(self._catalog.get_endpoint_by_type(Component.Types.BAREMETAL.value))
-        )
-        response.raise_for_status()
-        return response.json().get('conductors', [])
+        def use_legacy_conductors_resource(microversion):
+            self.log.debug("Configured ironic microversion: %s", microversion)
+            if not microversion:
+                return True
+            legacy_microversion = True
+            try:
+                legacy_microversion = float(microversion) < 1.43
+            except Exception as e:
+                if microversion.lower() == 'latest':
+                    legacy_microversion = False
+                else:
+                    raise Exception(f"Invalid ironic microversion, cannot collect baremetal conductors: {str(e)}")
+            self.log.debug(
+                "Collecting baremetal conductors with use_legacy_conductors_resource =%s", legacy_microversion
+            )
+            return legacy_microversion
+
+        ironic_endpoint = self._catalog.get_endpoint_by_type(Component.Types.BAREMETAL.value)
+
+        params = {}
+        url = '{}/v1/conductors'.format(ironic_endpoint)
+
+        return self.make_paginated_request(url, 'conductors', 'uuid', params=params)
```

Comment:
```suggestion
        return self.make_paginated_request(url, 'conductors', 'uuid', params={})
```
nit- this is a little cleaner

Reply:
Done.